Add compatibility for multi_json versions less than 1.3.0

adorr opened the pull request.

ileitch commented:
I think I'd rather support all versions. Can you instead use a `respond_to?(:dump)` and keep the `~> 1.0` dependency.

Also, out of curiosity, are you using rapns at Desk.com?

Commit:
Add compatability for multi_json versions less than 1.3.0

adorr commented:
@ileitch I've updated this accordingly. respond_to didn't work with load. That method attempts to load a file in pre-1.3.0 versions, so there's an explicit version check. I added a few tests as well.

We aren't currently using rapns in production, but I'll be happy to list us in the README if and when that happens.
